contracts/messaging/apps/BatchMessageSender.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+
+import "../framework/SynMessagingReceiver.sol";
+
+pragma solidity 0.8.13;
+
+/** @title Example app of sending multiple messages in one transaction
+ */
+
+contract BatchMessageSender is SynMessagingReceiver {
+    constructor(address _messageBus) {
+        messageBus = _messageBus;
+    }
+
+    function sendMultipleMessages(bytes32[] memory _receiver, uint256[] memory _dstChainId, bytes[] memory _message, bytes[] memory _options) public payable {
+        require(_receiver.length == _dstChainId.length);
+        require(_receiver.length == _message.length);
+        require(_receiver.length == _options.length);
+
+        uint256 feePerMessage = msg.value / _message.length;
```

Comment:
If the function takes an array of destination chain ids, one can assume it's possible to batch send messages to different chains. Which means that fees may vary for each msg in the batch, so using average will work only if sending to the same chain.